src/test/java/org/jenkins_ci/plugins/flexible_publish/FlexiblePublisherTest.java

```diff
@@ -158,4 +171,322 @@ public void testMultipleConditionsMultipleActions() throws Exception {
+            return BuildStepMonitor.BUILD;
+        }
+        
+        @Override
+        public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener)
+                throws InterruptedException, IOException {
+            throw new IOException("Intended abort");
+            //return true;
+        }
+        
+        @Extension
+        public static class DescriptorImpl extends BuildStepDescriptor<Publisher> {
+            @Override
+            public boolean isApplicable(Class<? extends AbstractProject> jobType) {
+                return true;
+            }
+            
+            @Override
+            public String getDisplayName() {
+                return "ThorwGeneralExceptionPublisher";
+            }
+        }
+    }
+    
+    public void testRunAllPublishers() throws Exception {
+        // Jenkins executes all publishers even one of them failed.
+        {
+            FreeStyleProject p = createFreeStyleProject();
+            
+            p.getBuildersList().add(new FileWriteBuilder("artifact.txt", "blahblahblah"));
+            p.getPublishersList().add(new FailurePublisher());
+            p.getPublishersList().add(new ArtifactArchiver("**/*", "", false));
+            
+            FreeStyleBuild b = p.scheduleBuild2(0).get();
+            assertBuildStatus(Result.FAILURE, b);
+            
+            // ArtifactArchiver is executed even prior publisher fails.
+            assertTrue(new File(b.getArtifactsDir(), "artifact.txt").exists());
+        }
+        
+        // Jenkins executes all publishers even one of them throws AbortException.
+        {
+            FreeStyleProject p = createFreeStyleProject();
+            
+            p.getBuildersList().add(new FileWriteBuilder("artifact.txt", "blahblahblah"));
+            p.getPublishersList().add(new ThorwAbortExceptionPublisher());
+            p.getPublishersList().add(new ArtifactArchiver("**/*", "", false));
+            
+            FreeStyleBuild b = p.scheduleBuild2(0).get();
+            assertBuildStatus(Result.FAILURE, b);
+            
+            // ArtifactArchiver is executed even prior publisher fails.
+            assertTrue(new File(b.getArtifactsDir(), "artifact.txt").exists());
+            
+            // Somehow Jenkins prints stacktrace for AbortException. You can see that here.
+            // System.out.println(b.getLog());
+        }
+        
+        // Jenkins executes all publishers even one of them throws any Exceptions.
+        {
+            FreeStyleProject p = createFreeStyleProject();
+            
+            p.getBuildersList().add(new FileWriteBuilder("artifact.txt", "blahblahblah"));
+            p.getPublishersList().add(new ThorwGeneralExceptionPublisher());
+            p.getPublishersList().add(new ArtifactArchiver("**/*", "", false));
+            
+            FreeStyleBuild b = p.scheduleBuild2(0).get();
+            assertBuildStatus(Result.FAILURE, b);

```

Comment:
Tried TestProxyPublisher but junit exceptions caught and not throwed. Have no idea how it can be mocked, so your variant with checking final build results can be enough.